src/expander_pipeline.jl

```diff
@@ -1,20 +1,25 @@
 # Defines node "expanders" that transform nodes from the parsed markdown files.
 
+function clear_global!(M::Module, name::Symbol)
+    isconst(M, name) && return
+    VERSION >= v"1.9" ? setglobal!(M, name, nothing) : Core.eval(M, :($name = $nothing))
+    nothing
+end
+
 # helper for "cleaning up" content of modules to enable garbage collection.
 # See also <https://github.com/JuliaDocs/Documenter.jl/issues/2640>.
 function clear_module!(M::Module)
     # we need `invokelatest` here for Julia >= 1.12 (or 1.13?)
-    for name in Base.invokelatest(names, M, all = true)
-        if !isconst(M, name)
-            # see, e.g https://github.com/JuliaDocs/Documenter.jl/issues/2673
-            # it is not possible to set `nothing` to variables, which are strongly typed
-            # still attempt to set it, but ignore any errors
-            try
-                @eval M $name = $nothing
-            catch err
-                @debug "Could not clear variable `$name` by assigning `nothing`" err
-            end
+    for name in Base.invokelatest(names, M, all = true)::Vector{Any}
+        # see, e.g https://github.com/JuliaDocs/Documenter.jl/issues/2673
+        # it is not possible to set `nothing` to variables, which are strongly typed
```

Comment:
Unrelated, but while we are it, isn't the grammar here a bit odd? Perhaps like this?
```suggestion
        # it is not possible to set variables which are strongly typed to `nothing`
```

Reply:
More related: is there a clean way for us to determine that a variable / binding is strongly typed?

Reply:
Sure. `Nothing <: Core.get_binding_type(M, name)`